src/attributes.js

```diff
@@ -82,6 +79,20 @@ export const hasMany = (resource, { name } = {}) => ({
   isArray: true,
 });
 
+const attribute = ({ name, isType, cast, defaultValue, nullable = false }) => ({
+  type: 'attribute',
+  name,
+  cast: (value) => {
+    if (value === undefined) {
+      return typeof defaultValue === 'function'
+        ? defaultValue.call(value)
+        : defaultValue && JSON.parse(JSON.stringify(defaultValue));
```

Comment:
Why do we need to stringify and parse?

If defaultValue is a string, it will wrap the string in (most likely unintended) quotes
```js
JSON.stringify('hello');
// ""hello""
```